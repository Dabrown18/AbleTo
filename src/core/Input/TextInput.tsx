import React, { ComponentProps, FunctionComponent, useState } from 'react';
import { StyleSheet } from 'react-native';
import { HelperText, TextInput as TextInputBase } from 'react-native-paper';

import colors from '@src/core/colors';

// Copying over the props from TextInput for now, until we create our own version
// interface will most likely match, so no need to type these out now
// (I'm also lazy)
type TextInputProps = Omit<ComponentProps<typeof TextInputBase>, 'error'>;
type Props = { label: string; error?: string } & Omit<TextInputProps, 'placeholder' | 'label'>;

const TextInput: FunctionComponent<Props> = ({ style, label, error, onFocus, onBlur, ...rest }) => {
    const [isFocused, setIsFocused] = useState(false);

    return (
        <>
            <TextInputBase
                onFocus={(e) => {
                    setIsFocused(true);
                    onFocus?.(e);
                }}
                onBlur={(e) => {
                    setIsFocused(false);
                    onBlur?.(e);
                }}
                outlineColor={colors.gray200}
                activeOutlineColor={colors.primary600}
                accessibilityLabel={label}
                label={label}
                error={!!error}
                // Hide the placeholder when the input is focused
                placeholder={isFocused ? undefined : label}
                theme={{ colors: { text: colors.black, placeholder: colors.gray800 } }}
                style={[style, styles.input]}
                dense
                {...rest}
            />

            <HelperText type="error" visible={!!error} accessibilityLiveRegion="assertive">
                {error}
            </HelperText>
        </>
    );
};

const styles = StyleSheet.create({
    input: {
        backgroundColor: colors.white,
    },
});

export default TextInput;
