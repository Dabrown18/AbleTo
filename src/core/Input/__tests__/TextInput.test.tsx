import { render } from '@testing-library/react-native';
import React from 'react';

import TextInput from '../../Input/TextInput';

test('TextInput snapshot', () => {
    const tree = render(<TextInput label="Type in me!" />).toJSON();

    expect(tree).toMatchSnapshot();
});
