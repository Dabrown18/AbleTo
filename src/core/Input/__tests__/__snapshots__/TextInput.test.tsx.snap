// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput snapshot 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgb(231, 231, 231)",
          "borderTopLeftRadius": 4,
          "borderTopRightRadius": 4,
        },
        Object {
          "backgroundColor": "#fff",
        },
      ]
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.26)",
          "bottom": 0,
          "height": 2,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "transform": Array [
            Object {
              "scaleY": 0.5,
            },
          ],
          "zIndex": 1,
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "paddingBottom": 0,
            "paddingTop": 0,
          },
          Object {
            "minHeight": 52,
          },
        ]
      }
    >
      <View
        collapsable={false}
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
            "zIndex": 3,
          }
        }
      >
        <Text
          collapsable={false}
          numberOfLines={1}
          onLayout={[Function]}
          style={
            Object {
              "color": "#6200ee",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "left": 0,
              "opacity": 0,
              "paddingLeft": 12,
              "paddingRight": 12,
              "position": "absolute",
              "textAlign": "left",
              "top": 28,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
                Object {
                  "scale": 1,
                },
              ],
              "writingDirection": "ltr",
            }
          }
        >
          Type in me!
        </Text>
        <Text
          collapsable={false}
          numberOfLines={1}
          style={
            Object {
              "color": "#4B4845",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "left": 0,
              "opacity": 0,
              "paddingLeft": 12,
              "paddingRight": 12,
              "position": "absolute",
              "textAlign": "left",
              "top": 28,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
                Object {
                  "scale": 1,
                },
              ],
              "writingDirection": "ltr",
            }
          }
        >
          Type in me!
        </Text>
      </View>
      <TextInput
        accessibilityLabel="Type in me!"
        activeOutlineColor="#85356F"
        editable={true}
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        outlineColor="#E1E1E1"
        placeholder=""
        placeholderTextColor="#4B4845"
        selectionColor="#6200ee"
        style={
          Array [
            Object {
              "flexGrow": 1,
              "margin": 0,
            },
            Object {
              "paddingLeft": 12,
              "paddingRight": 12,
            },
            Object {
              "height": 52,
            },
            Object {
              "paddingBottom": 2,
              "paddingTop": 22,
            },
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            false,
            Array [
              Object {},
            ],
          ]
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>,
  <Text
    accessibilityLiveRegion="assertive"
    collapsable={false}
    onLayout={[Function]}
    style={
      Object {
        "color": "#B00020",
        "fontFamily": "System",
        "fontSize": 12,
        "fontWeight": "400",
        "opacity": 0,
        "paddingHorizontal": 12,
        "paddingVertical": 4,
        "textAlign": "left",
        "transform": Array [],
        "writingDirection": "ltr",
      }
    }
  />,
]
`;
